lib/node_modules/@stdlib/random/base/mt19937/lib/factory.js

```diff
@@ -62,19 +62,32 @@ var MATRIX_A = 0x9908b0df;
 * // returns 822569775
 */
 function factory( seed ) {
+	var seedIsArray = false;
 	var state;
+	var mag01 = [ 0x0, MATRIX_A ];
 	var mti;
 	var mt = new Array( N );
 	var s;
+	var i;
+	var j;
+	var k;
 
 	if ( arguments.length ) {
-		if ( !isPositiveInteger( seed ) ) {
-			throw new TypeError( 'invalid input argument. Must provide a positive integer. Value: `' + seed + '`.' );
-		}
-		if ( seed > MAX_SEED ) {
-			throw new RangeError( 'invalid input argument. Must provide a positive integer less than the maximum signed 32-bit integer. Value: `' + seed + '`.' );
+		if ( Array.isArray( seed ) ) {
+			seedIsArray = true;
+			i = 1;
+			j = 0;
+			k = ( N > seed.length ) ? N : seed.length;
```

Comment:
Use the `@stdlib/math/base/special/max` pkg and then `k = max( N, seed.length );` in order to better convey intent.